File: jasmine/lib/jasmine/email_user.ex
defmodule Jasmine.EmailUser do
  use Ecto.Schema
  import Ecto.Changeset

  schema "email_users" do
    field :user_id, :integer
    field :name, :string
    field :email, :string
    field :password, :binary
    field :avatar, :string
    field :confirmed_at, :utc_datetime_usec
    field :locked_at, :utc_datetime_usec
    field :deleted_at, :utc_datetime_usec
    field :version, :integer
    field :updated_at, :utc_datetime_usec
    field :created_at, :utc_datetime_usec
  end

  @doc false
  def changeset(email_user, attrs) do
    email_user
    |> cast(attrs, [
      :user_id,
      :name,
      :email,
      :password,
      :avatar,
      :confirmed_at,
      :locked_at,
      :deleted_at,
      :version,
      :updated_at,
      :created_at
    ])
    |> validate_required([:user_id, :name, :email, :password, :avatar])
  end
end

File: dahlia/src/main/kotlin/com/github/saturn_siv/palm/dahlia/plugins/cms/models/EmailUser.kt
package com.github.saturn_siv.palm.dahlia.plugins.cms.models

import jakarta.persistence.*
import java.time.Instant

@Table(name = "email_users")
@Entity(name = "user.by-email")
class EmailUser(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    var id: Int,
    @Column(nullable = false)
    var realName: String,
    @Column(nullable = false)
    var email: String,
    @Column(nullable = false)
    var password: ByteArray,
    @Column(nullable = false)
    var salt: ByteArray,
    @Column(nullable = false)
    var avatar: String,
    @Column
    var confirmedAt: Instant?,
    @Column
    var deletedAt: Instant?,
    @Column(nullable = false)
    var version: Int,
    @Column(nullable = false)
    var updatedAt: Instant,
    @Column(nullable = false)
    var createdAt: Instant,
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id")
    var user: User,
)